source/OrderedTypes/SupLattice-SmallBasis.lagda

```diff
@@ -0,0 +1,151 @@
+Ian Ray 05\02\2024
+
+A Sup Lattice L that has suprema for family of size 𝓥 has a basis if there is a
+type B : 𝓥 and map β : B → L such that
+  β b ≤ x is 𝓥 small
+and
+  x = ⋁ ↓ᴮ x
+for all x. 
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K --exact-split #-}
+
+open import MLTT.Spartan
+open import UF.Equiv
+open import UF.Equiv-FunExt
+open import UF.EquivalenceExamples
+open import UF.FunExt
+open import UF.Hedberg
+open import UF.Logic
+open import UF.Powerset-MultiUniverse
+open import UF.PropTrunc
+open import UF.Retracts
+open import UF.Sets
+open import UF.Subsingletons
+open import UF.Subsingletons-FunExt
+open import UF.SubtypeClassifier
+open import UF.Size
+open import UF.SmallnessProperties
+open import UF.UniverseEmbedding
+
+module OrderedTypes.SupLattice-SmallBasis
+       (pt : propositional-truncations-exist)
+       (fe : Fun-Ext)
+        where
+
+open import Locales.Frame pt fe hiding (⟨_⟩ ; join-of)
+open import Slice.Family
+open import UF.ImageAndSurjection pt
+open import OrderedTypes.SupLattice pt fe
+
+open AllCombinators pt fe
+open PropositionalTruncation pt
+
+\end{code}
+
+We define the notion of a small basis for a sup-lattice as well as some
+boiler plate. This consists of a type B and a map q : B → L. In a sense to be
+made precise we say the pair B and q generate the sup-lattice. This notion
+is crucial for the development of predicative order theory.
+
+\begin{code}
+
+module small-basis {𝓤 𝓦 𝓥 : Universe}
+                   {B : 𝓥  ̇}
+                   (L : Sup-Lattice 𝓤 𝓦 𝓥)
+                   (β : B → ⟨ L ⟩)
+                    where
+
+ _≤_ : ⟨ L ⟩ → ⟨ L ⟩ → Ω 𝓦
+ _≤_ = order-of L
+
+ ⋁_ : Fam 𝓥 ⟨ L ⟩ → ⟨ L ⟩
+ ⋁_ = join-of L
+
+ open Joins _≤_
+
+ ↓ᴮ : ⟨ L ⟩ → 𝓦 ⊔ 𝓥  ̇
+ ↓ᴮ x = Σ b ꞉ B , (β b ≤ x) holds
+
+ ↓ᴮ-to-base : (x : ⟨ L ⟩) → ↓ᴮ x → B
+ ↓ᴮ-to-base x = pr₁
+
+ ↓ᴮ-inclusion : (x : ⟨ L ⟩) → ↓ᴮ x → ⟨ L ⟩
```

Comment:
It's probably worth commenting that `↓ᴮ-inclusion` need not be an injection as `β` is not required to be injective.